### ISSUE TITLE
Authentication doesn't work in 0.8

### ISSUE THREAD

**Issue by jarrai** [MongoEngine/flask-mongoengine#259]
After I update version to 0.8, I get error 'not authorized on db to execute command',when I use version 0.75 it works OK.

**Comment by lafrech** [comment]
There are a few issues in 0.8 due to the rework of the connection module. Hopefully, they should be all fixed at once when this feature is fixed.

**Comment by ntcong** [comment]
As I said in https://github.com/MongoEngine/flask-mongoengine/issues/256#issuecomment-246181884, the original problem reported on #250

**Comment by citizen-stig** [comment]
I use the following workaround for flask:

```
app.before_first_request(
    lambda: db.connection.authenticate(app.config['MONGODB_SETTINGS'].get('username'),
                                       app.config['MONGODB_SETTINGS'].get('password')))
```

**Comment by trein** [comment]
It seems a pretty serious bug. Is `flask-mongoengine@0.8.0` usable without a fix? Before crunching on the source code, apart from @citizen-stig suggestion, is that another workaround for this?

**Comment by lafrech** [comment]
Contributors willing to fix the connexion regression in 0.8 are welcome to join in https://github.com/MongoEngine/flask-mongoengine/issues/71#issuecomment-252554167.

**Comment by wojcikstefan** [comment]
This issue should be addressed by #280

**Comment by wojcikstefan** [comment]
Should be fixed in v0.8.1, v0.8.2, and v0.9.0.
